### PR TITLE
[SYCL] Optionally print Graph to file

### PR DESCRIPTION
**Pull request by alexbatashev**
Add means to print Graph in dot format to file.
The following parameters may be passed to SYCL_PRINT_EXECUTION_GRAPH
env variable, that controls printer behavior:

- before_addCG -- print graph before addCG method
- after_addCG -- print graph after addCG method
- before_addCopyBack -- print graph before addCopyBack method
- after_addCopyBack -- print graph after addCopyBack method
- before_addHostAcc -- print graph before addHostAccessor method
- after_addHostAcc -- print graph after addHostAccessor method
- always -- print graph before and after each of the above methods

Multiple values may be passed like this:
SYCL_PRINT_EXECUTION_GRAPH="before_addCG,before_addCopyBack"

Signed-off-by: Alexander Batashev <alexander.batashev@intel.com>

### CODE REVIEW

**Comment by keryell** [comment]
Is this documented somewhere?
That sounds like a super cool feature of this tool chain you should communicate about!

**Comment by alexbatashev** [comment]
> Is this documented somewhere?
> That sounds like a super cool feature of this tool chain you should communicate about!

I’ve described variable [here](https://github.com/intel/llvm/pull/424/files#diff-64418657577445753512ac34089c9254). It is pretty straightforward. You set the variable, run your app, see .dot files in the working directory. Then you can convert it to svg/png/ etc with graphviz or use it with another tools. What other kind of documentation do you need?

File: sycl/source/detail/scheduler/commands.cpp [hunk 1]
```diff
@@ -273,6 +432,58 @@ cl_int MemCpyCommandHost::enqueueImp() {
...
+  }
+}
+
+void ExecCGCommand::printDot(std::ostream &Stream) const {
+  Stream << "\"" << this << "\" [style=filled, fillcolor=\"#AFFF82\", label=\"";
+
+  Stream << "ID = " << this << "\n";
+  Stream << "EXEC CG ON " << deviceToString(MQueue->get_device()) << "\\n";
+
+  if (MCommandGroup->getType() == detail::CG::KERNEL) {
```
Comment (romanovvlad) [inline sycl/source/detail/scheduler/commands.cpp]: Could you please use switch here?

**Comment by alexbatashev** [comment]
> Please apply clang-format to your patch.

Done

**Pull request merged**